### ISSUE TITLE
Nirspec wavelength units

### ISSUE THREAD

**Issue by nden** [STScI-JWST/jwst#155]
Make sure all Nirspec modes return the wavelength in microns.

**Comment by brechmos-stsci** [comment]
@nden and I talked about the ticket and this is my understanding (which might need to be modified).  

All the transforms in assign_wcs, for NIRSPEC, expect the wavelength data to be in meters, which it is currently. The output wavelength dataset is expected to be in microns, which it isn't (hence the ticket). So, I need to come up with a way to convert the wavelength units from meters to microns.

@nden and I discussed two possible methods:

1)  Add a transform using Identity and Constant so it would be **something** like:

```
(ra, dec, lambda) | Mapping(0,1,2,2) & Identity(2) * ( Identity(1) * Constant(10^6))
```
The problem is the inverse transform may take some work given the number of inputs is different.

2) Extend the Affine() Transform to move from 2D to 3D (or add a 3D) 

We could put it in jwst.transforms for now and it could be built primarily for this purpose. In this case we would need to create an Affine3D in which the diagonal would be (1, 1, 10^6).  I would also have to look at the inverse transform and we would have to be careful as the inverse transform would still do rotate-translate which is not the inverse of rotate-translate.

@nden and I talked and 2. might be the better option as it would be cleaner.

In either case the change should be incorporated into the v2v3 frame as opposed to adding a new frame only for the wavelength unit conversion.

**Comment by nden** [comment]
:+1:

**Comment by brechmos-stsci** [comment]
@nden as an aside, given the values have units attached to them is there not a "built-in" unit conversion utility?  Would something like that make more sense to pursue as it would be more generally applicable?

I am about 50% done for this ticket either way.

**Comment by nden** [comment]
`astropy.units` is generally used for unit conversion but models don't work with units yet although they will in the future when one would be able to pass `astropy.Quantity` to models. Currently `gwcs.coordinate_frame` has a method for attaching units to the output values. In the pipeline for now we need to return results in microns by default.

**Comment by nden** [comment]
Fixed in #243. Thanks @brechmos-stsci !
